### DISCUSSION TITLE
Create a disk image that boots and runs the app when the disk is inserted?

### DISCUSSION THREAD

**Discussion by muimota** [autc04/Retro68#117] [Q&A]
Is it possible to create with retro68 a disk image that boots and runs the app when the disk is inserted? 
  Just the way some games like [McFlip](https://macintoshgarden.org/games/mcflip) that when the computer boots with the disk inserted the game runs.

**Comment by autc04** [comment]
Yes it is...
The `LaunchAPPL` tool does exactly that when launching an app via miniVMac - it creates a bootable disk image and launches miniVMac.
Unfortunately, the code currently just puts the generated disk image in a generated directory and starts miniVMac right away.

It's probably possible to steal the disk image from that temporary directory before it gets deleted again, or you could have a look at what the source code does (especially `LaunchAPPL/Client/MiniVMac.cc`) and adapt it to your needs.

**Comment by muimota** [comment]
I found this other way, maybe it is obvious to classic mac users but it might be useful to not so experienced users.

1. Copy your executable into a *bootable* disk, a disk that has a folder named **System Folder**  with **System** and **Finder**
![image](https://user-images.githubusercontent.com/6080740/104013718-21eada00-51b2-11eb-98cb-fd1dead3ba62.png)
2. Select you executable and click on **Set Startup** in the **Special** Menu
![image](https://user-images.githubusercontent.com/6080740/104014101-c836df80-51b2-11eb-83d0-489e1f7297ee.png)
3. Now the app will launch when disk image is inserted. I guess this will also work on a real mac.

**Reply by autc04** [reply]
Indeed, I should have remembered to answer a bit more than the literal question of how to do it *with Retro68* :-)
